geoserver_init/index.js

```diff
@@ -0,0 +1,179 @@
+ */
+import GeoServerRestClient from 'geoserver-node-client';
+import {framedBigLogging, framedMediumLogging} from './js-utils/logging.js';
+import dockerSecret from './js-utils/docker-secrets.js';
+
+const verbose = process.env.GSINIT_VERBOSE;
+
+const geoserverUrl = process.env.GSPUB_GS_REST_URL || 'http://geoserver:8080/geoserver/rest/';
+const geoserverDefaultUser = process.env.GSINIT_GS_REST_DEFAULT_USER || 'admin';
+const geoserverDefaultPw = process.env.GSINIT_GS_REST_DEFAULT_PW || 'geoserver';
+
+verboseLogging('GeoServer REST URL: ', geoserverUrl);
+verboseLogging('GeoServer Default REST User:', geoserverDefaultUser);
+verboseLogging('GeoServer Default REST PW:  ', geoserverDefaultPw);
+
+const workspacesList = process.env.GSINIT_WS || 'station_data,image_mosaics';
+const stationWorkspace = process.env.GSINIT_STATION_WS || 'station_data';
+const stationDataStore = process.env.GSINIT_STATION_DS || 'station_data';
+const pgHost = process.env.GSINIT_PG_HOST || 'db';
+const pgPort = process.env.GSINIT_PG_PORT || '5432';
+const pgUser = process.env.GSINIT_PG_USER || 'app';
+const pgPassword = dockerSecret.read('app_password') || process.env.GSINIT_PG_PW;
+const pgSchema = process.env.GSINIT_PG_SCHEMA || 'station_data';
+const pgDb = process.env.GSINIT_PG_DB || 'sauber_data';
+const newGeoserverUser = dockerSecret.read('geoserver_user');
+const newGeoserverPw = dockerSecret.read('geoserver_password');
+const role = 'ADMIN';
+
+verboseLogging('-----------------------------------------------');
+
+verboseLogging('Workspaces:  ', workspacesList);
+verboseLogging('Station WS:  ', stationWorkspace);
+verboseLogging('Station DS:  ', stationDataStore);
+verboseLogging('PG Host:     ', pgHost);
+verboseLogging('PG Port:     ', pgPort);
+verboseLogging('PG User:     ', pgUser);
+verboseLogging('PG Schema:   ', pgSchema);
+verboseLogging('PG Database: ', pgDb);
+
+/**
+ * Main process:
+ *  - Create workspaces
+ *  - Change user + password
+ *  - Create store and layer for stations
+ */
+async function initGeoserver() {
+  framedBigLogging('Start initalizing SAUBER GeoServer...');
+
+  await adaptSecurity();
+
+  await createWorkspaces();
+
+  await createPostgisDatastore();
+
+  framedBigLogging('... DONE initalizing SAUBER GeoServer');
+}
+
+/**
+ * Adapts security settings for GeoServer
+ */
+async function adaptSecurity() {
+  const user = newGeoserverUser;
+  const userPw = newGeoserverPw;
+
+  if (!user || !userPw || user === '' || userPw === '') {
+    exitWithErrMsg('No valid user or user password given - EXIT.');
+  }
+
+  const userCreated =  await grc.security.createUser(user, userPw);
+  if (userCreated) {
+    console.info('Successfully created user', user);
+  }
+
+  const roleAssigend = await grc.security.associateUserRole(user, role);
+  if (roleAssigend) {
+    console.info(`Successfully added role ${role} to user ${user}`);
+  }
+
+  // disable user
+  const adminDisabled = await grc.security.updateUser('admin', 'geoserver', false);
```

Comment:
`admin` and `geoserver` should be replaced by the respective variables that are declared at the beginning of the document